src/app/component/MyAccountMyOrders/MyAccountMyOrders.component.js

```diff
@@ -13,15 +13,251 @@ import { PureComponent } from 'react';
 import PropTypes from 'prop-types';
 import './MyAccountMyOrders.style';
 
+import TextPlaceholder from 'Component/TextPlaceholder';
+import { Link } from 'react-router-dom';
+import { formatCurrency } from 'Util/Price';
+
+export const ORDERS_PATH = 'orders-list';
+
+const MY_ACCOUNT_PREFIX = 'my-account';
+const CREATED_AT = 'created_at';
+const GRAND_TOTAL = 'grand_total';
+const ID = 'increment_id';
+const STATUS_LABEL = 'status_label';
+const SHOW_MORE = 'show_more';
+
 class MyAccountMyOrders extends PureComponent {
     static propTypes = {
-        // TODO: implement prop-types
+        getOrderById: PropTypes.func.isRequired,
+        getOrderList: PropTypes.func.isRequired,
+        getFormattedDate: PropTypes.func.isRequired,
+        orderId: PropTypes.oneOfType([
+            PropTypes.string, PropTypes.number
+        ]).isRequired,
+        orderList: PropTypes.arrayOf(PropTypes.shape).isRequired,
+        isMobileTab: PropTypes.bool
+    };
+
+    static defaultProps = {
+        isMobileTab: false
     };
 
+    static getTabLabel() {
+        return __('Order history');
+    }
+
+    static getTabPathname() {
+        return ORDERS_PATH;
+    }
+
+    static isVisible() {
+        return true;
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.shownOrderValues = [
+            'increment_id',
+            'status_label',
+            'created_at',
+            'grand_total'
+        ];
+
+        this.tableHeadingRow = {
+            [ID]: `${__('Order')} №`,
+            [CREATED_AT]: __('Date'),
+            [STATUS_LABEL]: __('Status Label'),
+            [GRAND_TOTAL]: __('Order - Total'),
+            [SHOW_MORE]: __('Action')
+        };
+
+        this.renderOrderTableRow = this.renderOrderTableRow.bind(this);
+    }
+
+    componentDidMount() {
+        const {
+            getOrderList, getOrderById, isMobileTab, orderId
+        } = this.props;
+
+        if (isMobileTab) return;
+
+        if (orderId === ORDERS_PATH) {
+            getOrderList();
+        } else {
+            getOrderById(orderId);
+        }
+    }
+
+    processOrderItem(key, value, is_b2b) {
+        if (!this.shownOrderValues.includes(key)) {
+            return null;
+        }
+
+        switch (key) {
+        case CREATED_AT:
+            const { getFormattedDate } = this.props;
+            const formattedDate = getFormattedDate(value);
```

Comment:
Move this to sparate function each, better create a mapping, otherwise this is literally unusable. And very hard to extend.